pkg/kv/kvserver/kvserverpb/proposer_kv.proto

```diff
@@ -258,6 +252,22 @@ message ReplicatedEvalResult {
   // application while ensuring correctness in the case where the lease transfer
   // is applied on the new leaseholder through a Raft snapshot.
   kv.kvserver.readsummary.ReadSummary prior_read_summary = 22;
+  
+  // LinkExternalSSTable, like AddSSTable, is a side effect that must execute
+  // before the Raft application is committed. It must be idempotent to account
+  // for an ill-timed crash after applying the side effect, but before
+  // committing the batch.
```

Comment:
to be transparent: we considered reducing the scope of this pr to be only above raft: i.e. cmd_link_external_sstable would create a addsstable replicated eval result. One reason we want to plumb the logic below raft is because it makes it more clear that regular addsstable requests _always_ have data to sideload.